File: softgoals/models/small.pl

start     <-- [and([must(built), must(deployed)]),   % hard side: all gate
               and([cheap, fast, private])].         % soft side: label all
built     <-- [or([buy, diy])].
buy       <-- [vendor, breaks(cheap), helps(fast)].
diy       <-- [coders, helps(cheap), hurts(fast)].
deployed  <-- [or([cloud, onprem])].
cloud     <-- [helps(fast), hurts(private)].
onprem    <-- [makes(private), hurts(fast)].
usable    <-- [tested, helps(fast)].

File: softgoals/models/small.py
# small.py : buy-vs-build, in the algebra dialect.
from nfr5 import *

(built,deployed,buy,diy,vendor,coders,tested,usable,
 cheap,fast,private,cloud,onprem) = atoms(
  "built deployed buy diy vendor coders tested usable "
  "cheap fast private cloud onprem")

built    <= buy + diy
buy      <= vendor * breaks(cheap) * helps(fast)
diy      <= coders * helps(cheap) * hurts(fast)
deployed <= cloud + onprem
cloud    <= helps(fast) * hurts(private)
onprem   <= makes(private) * hurts(fast)
usable   <= tested * helps(fast)

HARD = [built, deployed]
SOFT = cheap + fast + private
